pkg/registry/core/service/ipallocator/allocator.go

```diff
@@ -80,8 +83,13 @@ type Range struct {
 
 // NewAllocatorCIDRRange creates a Range over a net.IPNet, calling allocatorFactory to construct the backing store.
 func NewAllocatorCIDRRange(cidr *net.IPNet, allocatorFactory allocator.AllocatorFactory) (*Range, error) {
-	max := RangeSize(cidr)
-	base := bigForIP(cidr.IP)
+	max := utilnet.RangeSize(cidr)
+	ones, bits := cidr.Mask.Size()
```

Comment:
this ipv6-specific logic denormalized in several places is not easy to understand without the accompanying comment. Also, do we even need to detect whether this is ipv6, or can we just limit the max to 65536, like this:

```go
// The max size will be limited to 65536.
// This is due to the allocator keeping track of all the	
// allocated IP's in a bitmap. This will keep the size of	
// the bitmap to 64k.
max := utilnet.RangeSize(cidr)
if max > 65536 {
  max = 65536
}
```

Reply:
> But makes `TestAllocate` fail due to an overflow where previously we were returning `65536`.

That's a little concerning. Does that mean callers of RangeSize cannot safely use or understand the result?

Reply:
this means that callers should always check boundaries. I'd either:

- check boundaries first then fallback on `RangeSize()` call
- bake the boundaries check into k/utils

would defer to @thockin on this one

Reply:
an API that requires callers to do out of bounds checks sounds really problematic

Reply:
if there's a possible for overflow, I'd have expected the method signature to return a defined error

Reply:
Agreed, let's go with bound checks in k/utils